### ISSUE TITLE
Yosys uhdm frontend is creating modules on usage

### ISSUE THREAD

**Issue by kamilrakoczy** [alainmarcel/uhdm-integration#48]
Yosys uhdm frontend doesn't distinguish usage of module from a definition and it is always creating module if it can't find one. Example:

```
image/bin/surelog -parse -sverilog tests/ibex/ibex/build/lowrisc_ibex_top_artya7_0.1/src/lowrisc_ibex_fpga_xilinx_shared_0/rtl/fpga/xilinx/clkgen_xil7series.sv
yosys/yosys -p "read_uhdm -debug slpp_all/surelog.uhdm"
```
UHDM Output:
```
    |vpiModule:
    \_module: work@clkgen_xil7series::BUFG (work@clkgen_xil7series.clk_fb_bufg) clkgen_xil7series.sv:63     : , parent:work@clkgen_xil7series
      |vpiDefName:work@clkgen_xil7series::BUFG
      |vpiName:clk_fb_bufg
      |vpiFullName:work@clkgen_xil7series.clk_fb_bufg
      |vpiPort:
      \_port: (I), parent:work@clkgen_xil7series.clk_fb_bufg
        |vpiName:I
        |vpiHighConn:
        \_ref_obj: (clk_fb_unbuf), line:64
          |vpiName:clk_fb_unbuf
          |vpiActual:
          \_logic_net: (@@BAD_SYMBOL@@), line:16, parent:work@clkgen_xil7series
      |vpiPort:
      \_port: (O), parent:work@clkgen_xil7series.clk_fb_bufg
        |vpiName:O
        |vpiHighConn:
        \_ref_obj: (clk_fb_buf), line:65
          |vpiName:clk_fb_buf
          |vpiActual:
          \_logic_net: (@@BAD_SYMBOL@@), line:15, parent:work@clkgen_xil7series
```
Yosys AST output:
```
    AST_MODULE <slpp_all/surelog.uhdm:0.0-0.0> [0x5574caf67130] str='\clkgen_xil7series::BUFG' basic_prep
       AST_WIRE <slpp_all/surelog.uhdm:0.0-0.0> [0x5574caf67f60] str='\I' basic_prep port=30 range=[0:0]
       AST_WIRE <slpp_all/surelog.uhdm:0.0-0.0> [0x5574caf680a0] str='\O' basic_prep port=31 range=[0:0]
```
``clkgen_xil7series::BUFG`` module was created, but it was just used in ``clkgen_xil7series.sv``.

**Comment by kamilrakoczy** [comment]
It is partially fixed in #64, modules are now created with correct names (and ports have correct directions), but we are still creating ``AST_MODULE`` node on usage.

**Comment by kamilrakoczy** [comment]
I fixed creating modules on usage in https://github.com/antmicro/yosys/commit/def67af4e11acd370e554004c48ebc97fb1d3da7. This fix do not remove modules that are defined (but not used) from the design. In this kind of modules, uhdm is missing information about ranges in ports and yosys is failing (on ``ibex_multdiv_slow.sv``) with message ``ERROR: Unsupported expression on dynamic range select on signal``. I created separate issue for this in Surelog https://github.com/alainmarcel/Surelog/issues/892
